src/test/java/org/minions/devfund/sergio/SquarelotronTest.java

```diff
@@ -0,0 +1,347 @@
+        final int size = 6;
+        final int[][] expected = {{1, 2, 3, 4, 5, 6}, {7, 26, 27, 28, 29, 12}, {13, 20, 15, 16, 23, 18},
+                {19, 14, 21, 22, 17, 24}, {25, 8, 9, 10, 11, 30}, {31, 32, 33, 34, 35, 36}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.upsideDownFlip(2)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 6x6.
+     */
+    @Test
+    public void testUpsideDownFlipIn6x6SquareRingThree() {
+        final int size = 6;
+        final int ring = 3;
+        final int[][] expected = {{1, 2, 3, 4, 5, 6}, {7, 8, 9, 10, 11, 12}, {13, 14, 21, 22, 17, 18},
+                {19, 20, 15, 16, 23, 24}, {25, 26, 27, 28, 29, 30}, {31, 32, 33, 34, 35, 36}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.upsideDownFlip(ring)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values of the processed square of the method upsideDownFlip.
+     */
+    @Test
+    public void testUpsideDownFlipInOverCharge() {
+        final int size = 5;
+        final int[][] expected = {{1, 2, 3, 4, 5}, {6, 17, 18, 19, 10}, {11, 12, 13, 14, 15},
+                {16, 7, 8, 9, 20}, {21, 22, 23, 24, 25}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.upsideDownFlip(2)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 2x2.
+     */
+    @Test
+    public void testMainDiagonalFlipIn2x2SquareRingOne() {
+        final int size = 2;
+        final int[][] expected = {{1, 3}, {2, 4}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.mainDiagonalFlip(1)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 6x6.
+     */
+    @Test
+    public void testMainDiagonalFlipIn6x6SquareRingOne() {
+        final int size = 6;
+        final int[][] expected = {{1, 7, 13, 19, 25, 31}, {2, 8, 9, 10, 11, 32}, {3, 14, 15, 16, 17, 33},
+                {4, 20, 21, 22, 23, 34}, {5, 26, 27, 28, 29, 35}, {6, 12, 18, 24, 30, 36}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.mainDiagonalFlip(1)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 4x4.
+     */
+    @Test
+    public void testMainDiagonalFlipIn4x4RingTwo() {
+        final int size = 4;
+        final int[][] expected = {{1, 2, 3, 4}, {5, 6, 10, 8}, {9, 7, 11, 12},
+                {13, 14, 15, 16}};
+        squarelotron = new Squarelotron(size);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)